### PR TITLE
User Satisfaction Index V1

### PR DESCRIPTION
**Pull request by gregglind**
It's time to test the first version of the USI. Here are the details:

URL: https://qsurvey.mozilla.com/s3/Firefox-User-Sentiment
Respondants: 1000ish
Channels: All channels
Distribution: EN*
Launch Date: 2016/01/22
Duration: A week max, less would be betterish?

### CODE REVIEW

File: test/recipes/test-heartbeat-by-user-first-impression.js [hunk 1]
```diff
@@ -465,4 +463,14 @@ describe("heartbeat-by-user-first-impression", function () {
       })
     });
   });
+
+
```
Comment (rayborn) [inline test/recipes/test-heartbeat-by-user-first-impression.js]: I can see the value in stubbing, but I'd leave a TODO

File: src/recipes/heartbeat-by-user-first-impression/utils.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+};
+
+var getEngagementUrl = function(locale, opts) {
+  let ans;
+  if (locale == "de") {
+    return null;
+  }
+  if (/^en-/.test(locale)) {
+    let ans = cutBreaks(surveyUrls['en'], surveyBreaks['en'])
+    if (ans)
```
Comment (rayborn) [inline src/recipes/heartbeat-by-user-first-impression/utils.js]: ?

File: src/recipes/heartbeat-by-user-first-impression/utils.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  }
+};
+
+var getEngagementUrl = function(locale, opts) {
+  let ans;
+  if (locale == "de") {
+    return null;
+  }
+  if (/^en-/.test(locale)) {
+    let ans = cutBreaks(surveyUrls['en'], surveyBreaks['en'])
```
Comment (rayborn) [inline src/recipes/heartbeat-by-user-first-impression/utils.js]: Shouldn't surveyUrls and surveyBreaks be parameters now that getEngagementUrl is in its own file?

File: src/recipes/heartbeat-by-user-first-impression/config.js [hunk 4]
```diff
@@ -33,24 +35,40 @@ let supportedLocales = [ 'de',
 ];
 
 module.exports = {
-  "nightly": {
-    restdays: 30,
-    sample: 1 / thousand,  // 1 of 1000
-    locales: supportedLocales
+  VERSION: VERSION,
+
+  surveyUrls: {
```
Comment (rayborn) [inline src/recipes/heartbeat-by-user-first-impression/config.js]: surveyUrl is currently assumed to be 1:1 with surveyBreaks.  Can we make these one data structure or add a TODO?
e.g.
surveys: {
  <locale>: ['url': <url>, 'break': <break>]
}

File: src/recipes/heartbeat-by-user-first-impression/index.js [hunk 5]
```diff
@@ -224,43 +233,9 @@ let run = function (state, extras) {
...
   //let engagementUrl =  `https://www.mozilla.org/en-US/firefox/feedback/?updateChannel=${state.updateChannel}&fxVersion=${state.fxVersion}`;  //"http://localhost/enagement.html",
+  let engagementUrl = getEngagementUrl(locale, {VERSION: VERSION, state:state});
 
-  let eUrls = [
-  ];
-
-  let cutBreaks = function (arr, breaks, rng=Math.random()) {
-    // should have tests, gah!  was not ready to deal with this yet.
-    // should have way of overring rng too
-    let out;
```
Comment (rayborn) [inline src/recipes/heartbeat-by-user-first-impression/index.js]: Looks like this isn't used here.  Probably a former return var.

File: src/recipes/heartbeat-by-user-first-impression/utils.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  indent:2, maxerr:50, devel:true, node:true, boss:true, white:true,
+  globalstrict:true, nomen:false, newcap:true, esnext: true, moz: true  */
+
+/*global require, exports, log */
+
+
+let cutBreaks = function (arr, breaks, rng=Math.random()) {
+  // should have tests, gah!  was not ready to deal with this yet.
+  // should have way of overring rng too
+  let out;
```
Comment (rayborn) [inline src/recipes/heartbeat-by-user-first-impression/utils.js]: Looks like this isn't used here. Probably a former return var.

**Comment by rayborn** [comment]
Some minor issues with utils and some suggestions in config.  The rest LGTM.

File: src/recipes/heartbeat-by-user-first-impression/config.js (hunk 4 above)
Comment (gregglind) [inline src/recipes/heartbeat-by-user-first-impression/config.js]: no I am not going to do this, because in JS it's really painful.  Design choice.

**Comment by gregglind** [comment]
@rayborn, moved all the tech debt to #221

**Pull request merged**